elastic/beats
# [Metricbeat] add collecting tags for rds metricset

## What does this PR do?

This PR adds tag collection in rds metricset by default.

## Why is it important?

Tags are important part of metadata for RDS instances. With this PR, users can collect metrics from RDS and use tags for filtering.

## Checklist


## How to test this PR locally

Create RDS with tags first, then use config below to start aws module:
```
- module: aws
  period: 5m
  credential_profile_name: elastic-beats
  metricsets:
    - rds
```

You should see tags collected in events, for example:
```
{
    "@timestamp": "2017-10-12T08:05:34.853Z",
    "aws": {
        "rds": {
            "aurora_bin_log_replica_lag": 0,
            "aurora_replica.lag_max.ms": 19.108999252319336,
            "aurora_replica.lag_min.ms": 19.108999252319336,
            "cache_hit_ratio.buffer": 100,
            "cache_hit_ratio.result_set": 0,
            "cpu": {
                "total": {
                    "pct": 0.04
                }
            },
            "database_connections": 0,
            "db_instance": {
                "arn": "arn:aws:rds:us-east-1:428152502467:db:database-1-instance-1",
                "class": "db.r5.large",
                "identifier": "database-1-instance-1",
                "status": "available"
            },
            "db_instance.identifier": "database-1-instance-1",
            "deadlocks": 0,
            "disk_usage": {
                "bin_log.bytes": 0
            },
            "engine_uptime.sec": 2704277,
            "free_local_storage.bytes": 31745863680,
            "freeable_memory.bytes": 4634234880,
            "latency": {
                "commit": 5.270933333333333,
                "ddl": 0,
                "delete": 0,
                "dml": 0.1624,
                "insert": 0.1624,
                "select": 0.17333862433862435,
                "update": 0
            },
            "login_failures": 0,
            "queries": 9.11833836203304,
            "throughput": {
                "commit": 0.5000916834753039,
                "ddl": 0,
                "delete": 0,
                "dml": 0.5000916834753039,
                "insert": 0.5000916834753039,
                "network": 1.4,
                "network_receive": 0.7,
                "network_transmit": 0.7,
                "select": 3.150577605894414,
                "update": 0
            },
            "transactions": {
                "active": 0,
                "blocked": 0
            }
        },
        "tags": {
            "created-by": "ks",
            "dept": "engr"
        }
    },
    "cloud": {
        "account": {
            "id": "428152502467",
            "name": "elastic-beats"
        },
        "availability_zone": "us-east-1b",
        "provider": "aws",
        "region": "us-east-1"
    },
    "event": {
        "dataset": "aws.rds",
        "duration": 115000,
        "module": "aws"
    },
    "metricset": {
        "name": "rds",
        "period": 10000
    },
    "service": {
        "type": "aws"
    }
}
```
Also test with `tags_filter`:
```
- module: aws
  period: 5m
  credential_profile_name: elastic-beats
  tags_filter:
      - key: "dept"
        value: "eng"
  metricsets:
    - rds
```
## Related issues

- Closes elastic/beats#16358
